algebird-core/src/main/scala/com/twitter/algebird/MapAlgebra.scala

```diff
@@ -187,10 +187,28 @@ object MapAlgebra {
   def removeZeros[K, V: Monoid](m: Map[K, V]): Map[K, V] =
     m filter { case (_, v) => Monoid.isNonZero(v) }
 
-  // groupBy ks, sum all the vs
+  /**
+   * For each key, sum all the values. Note that if V is a Monoid, the current 
+   * implementation will drop from the output any key where the values are all
+   * Monoid.zero. If the Semigroup is a Monoid, This function is equivalent to:
+   *
+   *   pairs.filter(_._2 != Monoid.zero).groupBy(_._1).mapValues(_.map(_._2).sum)
+   *
+   * Otherwise, the function is equivalent to:
+   *
+   *   pairs.groupBy(_._1).mapValues(_.map(_._2).sum)
+   */
   def sumByKey[K, V: Semigroup](pairs: TraversableOnce[(K, V)]): Map[K, V] =
     Monoid.sum(pairs map { Map(_) })
 
+  /**
+   * For each key, creates a list of all values. This function is equivalent to:
+   *
+   *   pairs.groupBy(_._1).mapValues(_.map(_._2))
+   */
+  def group[K, V](pairs: TraversableOnce[(K, V)]): Map[K, List[V]] =
+    sumByKey(pairs.map { case (k, v) => k -> List(v) })
```

Comment:
actually, a few monoids (`List[T]` and `String`) have O(N^2) `plus` operations (repeatedly concatenating is not efficient).

This is why we added `Batched[T]`. I think for a library function like this, we probably need a special purpose method.
Something like:
```scala
    if (items.isEmpty) Map.empty
    else {
      val mutable = MMap[K, V]()
      items.foreach { case (k, v) =>
          case (k, v) =>
            val oldVOpt = mutable.get(k)
            // sorry for the micro optimization here: avoiding a closure
            val bldr = if (oldVOpt.isEmpty) {
                  val b = List.newBuilder[V]
                  mutable.update(k, b)
                  b
               } else oldVOpt.get
            bldr += v
        }
        mutable.map { case (k, bldr) => (k, bldr.result) } : Map[K, List[V]]
      }

```